### PR TITLE
Use Preferences.jl for truncated field printing

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR attempts to use Preferences.jl instead of using an existing pattern used in a few places, where internally defined methods are expected to be (optionally) overloaded:

```julia
truncate_printing_field_types() = false
enable_threading() = false
allow_mismatched_diagonalized_spaces() = false
assert_exact_transform() = false
allow_mismatched_fd_spaces() = false
```

I'm not super familiar with Preferences.jl, but it may be a better way of handling these global / package options

### CODE REVIEW

**Comment by charleskawczynski** [comment]
bors try

File: src/Fields/Fields.jl [hunk 1]
```diff
@@ -459,17 +459,14 @@ long.
...
+import Preferences
+Preferences.@set_preferences!(Pair("TruncateClimaCoreFieldPrinting" => false))
 ```
 =#
-truncate_printing_field_types() = false
-
-function Base.show(io::IO, ::Type{T}) where {T <: Fields.Field}
-    if truncate_printing_field_types()
+import Preferences
+if Preferences.@load_preference("TruncateClimaCoreFieldPrinting", false)
```
Comment (simonbyrne) [inline src/Fields/Fields.jl:467]: Preferences are local to the package, so don't need to be qualified
```suggestion
if Preferences.@load_preference("TruncateFieldPrinting", false)
```

File: test/Fields/field.jl [hunk 2]
```diff
@@ -283,13 +283,14 @@ end
 end
 
 # Test truncated field type printing:
-ClimaCore.Fields.truncate_printing_field_types() = true
+import Preferences
+Preferences.@set_preferences!(Pair("TruncateClimaCoreFieldPrinting" => true))
```
Comment (simonbyrne) [inline test/Fields/field.jl:287]: This probably won't work, as it needs to be set before the module is loaded.

File: src/Fields/Fields.jl (hunk 1 above)
Comment (charleskawczynski) [reply src/Fields/Fields.jl:467]: Ah, good to know. That's what I originally had but thought it'd be safer to qualify. I'll change it back, thanks!

File: test/Fields/field.jl (hunk 2 above)
Comment (charleskawczynski) [reply test/Fields/field.jl:287]: Yeah, I wasn't sure but I thought I'd try. What's a good way to test this, then? Can I add a separate `@safetestset "Recursive" begin @time include("test_truncated_field_printing.jl") end`?

Comment (charleskawczynski) [reply test/Fields/field.jl]: Let me try that

**Comment by charleskawczynski** [comment]
bors try

File: test/Fields/field.jl (hunk 2 above)
Comment (simonbyrne) [reply test/Fields/field.jl]: no, it has to be in a whole new session.

Comment (simonbyrne) [reply test/Fields/field.jl]: (it will also trigger precompilation)

**Comment by charleskawczynski** [comment]
Julia 1.10 will support depth limited stack printing, so let’s just close this and wait for improvements.

**Pull request closed without merging**